### PR TITLE
[NEW] Allow user to change room description

### LINKED ISSUE #1320: [NEW] Allow the user to update "Description" of the room

### PR DESCRIPTION
**Pull request by RobOSatch**
@RocketChat/ios

The owner can now edit the description of the room. The only thing that bothers me a bit is that the API sends a text message to the room which simply says "[New Description Name]". In the web application, it says "has changed the description to [New Description Name]". Is that expected behavior? Or does it have to be fixed in the app/server side (API)? :) 

Closes #1320

### CODE REVIEW

File: Rocket.Chat/Controllers/Chat/ChannelInfoViewController.swift [hunk 1]
```diff
@@ -67,6 +68,18 @@ class ChannelInfoViewController: BaseViewController {
...
+        
+        if let currentUser = AuthManager.currentUser() {
+            if currentUser == subscription?.roomOwner {
+                let saveButton = UIBarButtonItem(title: "Save", style: .plain, target: self, action: #selector(saveButtonTapped))
+                navigationItem.rightBarButtonItems?.append(saveButton)
+                self.saveButton = saveButton
+                self.saveButton?.isEnabled = false
+            }
+        }
+
```
[the remark this answers is not in the document]
Comment (RobOSatch) [reply Rocket.Chat/Controllers/Chat/ChannelInfoViewController.swift]: Okay, will do! When I pulled the code a lot of those warnings were already there, so I was assuming that I should just maintain the coding style, but I can remove the extra lines so the warnings go away!

**Comment by rafaelks** [comment]
@RobOSatch Nice one, thank you! Can you resolve the Hound issues please? Regarding the message, AFAIK the message is sent via the back-end, but I'll investigate it.

**Comment by RobOSatch** [comment]
@rafaelks Hound issues are resolved :) Yes. it's probably the backend sending the message. The work is currently on my wip-branch since I opened the PR with that one. Is it possible to change the branch to one with no "wip" tag? If I understood correctly then my code is ignored for merges as long as it has this tag.

**Comment by rafaelks** [comment]
Branch name is fine @RobOSatch, we'll review it for 2.2.0 (next release). Thanks for fixing the Hound issues. 👍

File: Rocket.Chat/Managers/Model/SubscriptionManager.swift [hunk 2]
```diff
@@ -339,6 +339,34 @@ extension SubscriptionManager {
...
+            guard !response.isError() else { return Log.debug(response.result.string) }
+            completion(response)
+        }
+    }
+
+    static func updateRoomTopic(subscription: Subscription, topic: String, completion: @escaping MessageCompletion) {
+        let request = [
+            "msg": "method",
+            "method": "saveRoomSettings",
+            "id": "16",
```
Comment (rafaelks) [inline Rocket.Chat/Managers/Model/SubscriptionManager.swift]: This param can be removed, the `id` is automatically added on `SocketManager.send()`

File: Rocket.Chat/Managers/Model/SubscriptionManager.swift [hunk 3]
```diff
@@ -339,6 +339,34 @@ extension SubscriptionManager {
             completion(response)
         }
     }
+
+    static func updateRoomDescription(subscription: Subscription, description: String, completion: @escaping MessageCompletion) {
+        let request = [
+            "msg": "method",
+            "method": "saveRoomSettings",
+            "id": "16",
```
Comment (rafaelks) [inline Rocket.Chat/Managers/Model/SubscriptionManager.swift]: This param can be removed, the `id` is automatically added on `SocketManager.send()`

**Comment by rafaelks** [comment]
@cardoso Updated it with `develop` branch. 👍

**Comment by rafaelks** [comment]
@RobOSatch Closing this one for now, we're going to rewrite this screen soon. Thank you very much for your PR! 👍

**Pull request closed without merging**